keadm/cmd/keadm/app/cmd/edge/join.go

```diff
@@ -109,21 +118,150 @@ func addJoinOtherFlags(cmd *cobra.Command, joinOptions *types.JoinOptions) {
 	cmd.Flags().StringVarP(&joinOptions.CertPort, types.CertPort, "s", joinOptions.CertPort,
 		"The port where to apply for the edge certificate")
 
+	cmd.Flags().StringVarP(&joinOptions.QuicPort, types.QuicPort, "q", joinOptions.QuicPort,
+		"The port where to apply for the edge quic")
+
+	cmd.Flags().StringVarP(&joinOptions.TunnelPort, types.TunnelPort, "n", joinOptions.TunnelPort,
+		"The port where to apply for the edge stream connected tunnel")
+
 	cmd.Flags().StringVar(&joinOptions.TarballPath, types.TarballPath, joinOptions.TarballPath,
 		"Use this key to set the temp directory path for KubeEdge tarball, if not exist, download it")
 
 	cmd.Flags().StringSliceVarP(&joinOptions.Labels, types.Labels, "l", joinOptions.Labels,
-		`use this key to set the customized labels for node. you can input customized labels like key1=value1,key2=value2`)
+		`Use this key to set the customized labels for node. you can input customized labels like key1=value1,key2=value2`)
+
+	cmd.Flags().BoolVarP(&joinOptions.EnableDefaultTaint, types.EnableDefaultTaint, "w", joinOptions.EnableDefaultTaint,
+		"To determine whether to add NoSchedule taint to the edge node")
+
+	cmd.Flags().BoolVarP(&joinOptions.EnableMetaServer, types.EnableMetaServer, "m", joinOptions.EnableMetaServer,
+		"To determine whether to enable metaServer component on edge node")
+
+	cmd.Flags().BoolVarP(&joinOptions.EnableServiceBus, types.EnableServiceBus, "k", joinOptions.EnableServiceBus,
+		"To determine whether to enable serviceBus component on edge node")
+
+	cmd.Flags().StringVarP(&joinOptions.DownloadRegion, types.DownloadRegion, "z", joinOptions.DownloadRegion,
+		"To distinguish download links for KubeEdge EdgeCore and Services, could be 'zh' or 'en'")
+
+	cmd.Flags().StringVarP(&joinOptions.KubeEdgeDownloadURL, types.KubeEdgeDownloadURL, "d", joinOptions.KubeEdgeDownloadURL,
+		"The custom download url for KubeEdge tar file and checksum file. i.e: https://github.com/kubeedge/kubeedge/releases/download")
+
+	cmd.Flags().StringVarP(&joinOptions.ServiceFileURLFormat, types.ServiceFileURLFormat, "c", joinOptions.ServiceFileURLFormat,
+		"The custom download url format for service file. i.e: https://raw.githubusercontent.com/kubeedge/kubeedge/release-%s/build/tools/%s, "+
+			"the first %s means the version number like 1.8, the second %s means the service name like edgecore.service")
 }
 
 // newJoinOptions returns a struct ready for being used for creating cmd join flags.
 func newJoinOptions() *types.JoinOptions {
```

Comment:
now, we first read key-value pairs from this file, and the command flag will overwrite those pairs? Is this a bit repetitive?

Reply:
I think we can use FTP to send these common parameters, thus the installation of the edge core can be simplified. One day, we can use scripts to join massive edge nodes.